objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionConfigHandler.java

```diff
@@ -346,11 +357,30 @@ boolean createImageRetriever(Map<String, ?> dataSourceConfig) {
     private boolean prepareCommunication(Map<String, ?> general) {
         int polling = -1; // initializing to an invalid input
         boolean queryable = false;
+        int maxRunningThreads = MAX_RUNNING_THREADS;
+        int maxQueuedTasks = MAX_QUEUED_TASKS;
+        
+        if (general.get("maxRunningThreads") instanceof Integer && (Integer) general.get("maxRunningThreads") > 0) {
+            maxRunningThreads = (Integer) general.get("maxRunningThreads");
+        }
+        
+        if (general.get("maxQueuedTasks") instanceof Integer && (Integer) general.get("maxQueuedTasks") > 0) {
+            maxQueuedTasks = (Integer) general.get("maxQueuedTasks");
+        }
+                
+        source.pool = new ThreadPoolExecutor(maxRunningThreads, maxRunningThreads, 0l, TimeUnit.MILLISECONDS, 
+                new LinkedBlockingQueue<Runnable>(maxQueuedTasks), new ThreadPoolExecutor.DiscardOldestPolicy());
```

Comment:
Should we document that if you DON'T want the background execution, you should set the maxThreads & maxQueud to 1, 0, respectively?  But that will generate errors, so we really don't have a BW compat way.  Which is probably OK with our single user, but we have to start thinking about all this...

Reply:
I agree that we should document this. The user would have to set both of those values to 1, since setting either one to 0 would throw an exception according to the documentation.

This enhancement will change the default behavior of the Object Rec source  to always process images in parallel, unless otherwise specified.

Reply:
I suspect (as noted above) that parallel (albeit limited) is now the only possibility.  That's probably fine, but we should verify that that's true & document.